apps/web/src/containers/App/App.tsx

```diff
@@ -56,12 +61,29 @@ import {
 } from '../../containers/TradingContest';
 import { Markets } from '../Markets';
 
+const RedirectHandler = () => {
+  const { pathname } = useLocation();
+  const history = useHistory();
+  const betaAccess = useNftContract();
+  const onboardWallet = getFromLocalStorage('onboard.js:last_connected_wallet');
+
+  useEffect(() => {
+    if (
+      (betaAccess === BETA_ACCESS.REJECTED && !pathname.includes('/contest')) ||
+      (onboardWallet.length === 0 && !pathname.includes('/contest'))
+    ) {
+      history.push('/contest');
+    }
+  }, [pathname, betaAccess, history, onboardWallet]);
+};
+
 const AllRoutes = () => {
   const [routeKey, setRouteKey] = useState('');
   // Have this hook here to ensure that all children routes will see updates if the onboard
   // context changes (there is a useEffect hook inside here listening for changes in the
   // onboard context)
   useConnect();
+  RedirectHandler();
```

Comment:
shouldn't this be titled like a hook: `useContestRedirect` or something like that?